app/src/main/java/com/epfl/beatlink/viewmodel/auth/FirebaseAuthViewModel.kt

```diff
@@ -80,6 +80,20 @@ class FirebaseAuthViewModel(private val authRepository: FirebaseAuthRepository)
     }
   }
 
+  fun signOut(onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
+    viewModelScope.launch {
```

Comment:
Here the `onSuccess` and `onFailure` parameters are not necessary as you already have these in the repository and also to keep it coherent with the other functions

Reply:
I actually need it to prevent the UI to navigate to the success screen even when it fails in the repo.